### PR TITLE
[inih] Upgrade from r51 to r56

### LINKED ISSUE #118: Memory allocation
For embedded systems it is convenient to use a custom memory allocator.

Thanks for the project!

Best Regards,
Aleksandr (aka bullit master)

### PR DESCRIPTION
**Pull request by SamuelMarks**
**Describe the pull request**
https://github.com/benhoyt/inih/releases

56

> Two updates:
> 
>   https://github.com/benhoyt/inih/pull/137
>   https://github.com/benhoyt/inih/pull/140

55

> Added "version" to meson.build config: https://github.com/benhoyt/inih/pull/135 (but bumped up to 55 in a subsequent commit, for this release).

54

> Mainly https://github.com/benhoyt/inih/pull/134, adding the visibility symbols to the Meson build config, but also [other small tweaks](https://github.com/benhoyt/inih/compare/r53...r54) to tests and so on.

53

> Updates to Meson config:
> 
>   https://github.com/benhoyt/inih/pull/124 meson: optionally depend on C++
>   https://github.com/benhoyt/inih/pull/125 meson: enable distro settings by default
>   https://github.com/benhoyt/inih/pull/126 meson: add static compile args to inih_dep

52
> Add INI_CUSTOM_ALLOCATOR to allow using a custom memory allocator. Per the README:
> 
>     By default when using the heap, the standard library's malloc, free, and realloc functions are used; to use a custom allocator, specify -DINI_CUSTOM_ALLOCATOR=1 (and -DINI_USE_STACK=0). You must define and link functions named ini_malloc, ini_free, and (if INI_ALLOW_REALLOC is set) ini_realloc, which must have the same signatures as the stdlib.h memory allocation functions.
> 
> See [tests/unittest_alloc.c](https://github.com/benhoyt/inih/blob/master/tests/unittest_alloc.c) for an example.
> 
> Fixes https://github.com/benhoyt/inih/issues/118.

- #### What does your PR fix?

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/inih/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
+project(iniparser LANGUAGES C CXX)
 
 set(SOURCES ini.c cpp/INIReader.cpp)
 set(HEADERS ini.h cpp/INIReader.h)
 
-add_library(inih ${SOURCES})
+add_library(inih ${HEADERS} ${SOURCES})
 
 install(
     TARGETS inih
```
Comment (JonLiu1993) [inline ports/inih/CMakeLists.txt:10]: @SamuelMarks ,Thanks for your pr, could you please export the config.cmake file?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/inih/CMakeLists.txt (hunk 1 above)
Comment (SamuelMarks) [reply ports/inih/CMakeLists.txt]: Did something along those lines, does it need to be `unofficial`?

Comment (JonLiu1993) [reply ports/inih/CMakeLists.txt]: YES

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Tested the usage successfully

**Comment by BillyONeal** [comment]
This is becoming a relatively large build system replacement. Do you know why we aren't just using upstream's meson one?

**Comment by SamuelMarks** [comment]
My main goal was debugging why it was getting my `#define`'d or `target_compile_definitions` params so I updated the port you already had in order to fix it. No opinion about build systems (us or theirs).

File: ports/inih/portfile.cmake [hunk 2]
```diff
@@ -3,20 +3,20 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO benhoyt/inih
-    REF d7f465792c0c7686b50ed45c9a435394ae418d3e # r51
-    SHA512 f3b621225933d74e9b130957aab1d55a82f06fd836aace439f9192a393d82a446e9f2152bb617a73a1dc681a821fc31d217c41f38c8ef8eeef7ad64c3e274b5d
+    REF 5e1d9e2625842dddb3f9c086a50f22e4f45dfc2b # r56
+    SHA512 477a66643f6636a5826a1206c6588a12827e24a4a2609e11f0695888998e2bfcba8bdb2240c561404ee675bf4c72e85d7d008a1fbddb142c0d263b413de8d358
     HEAD_REF master
 )
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
```
Comment (Thomas1664) [inline ports/inih/portfile.cmake:11]:
```suggestion
file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
```

Please put paths in quotes (same below).

File: ports/inih/portfile.cmake [hunk 3]
```diff
@@ -3,20 +3,20 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
```
Comment (Thomas1664) [inline ports/inih/portfile.cmake:17]:
```suggestion
vcpkg_cmake_install()
vcpkg_cmake_config_fixup()
```

Ports should use `vcpkg_cmake_config_fixup` when exporting cmake config.

File: ports/inih/portfile.cmake [hunk 4]
```diff
@@ -3,20 +3,20 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
 
 file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/inih RENAME copyright)
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share"
+                    "${CURRENT_PACKAGES_DIR}/debug/include")
```
Comment (Thomas1664) [inline ports/inih/portfile.cmake:22]: You should replace `inih` with `${PORT}` in the line above to ensure that the location copyright file aligns with the port name.

**Review by Thomas1664 — COMMENTED** [review]
I would like to see that `vcpkg_cmake_config_fixup` is used + quotes around paths.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
I agree with @Thomas1664 's comments

**Comment by BillyONeal** [comment]
(I tried to push a fix for you but:
image
)

**Comment by SamuelMarks** [comment]
I'll manually make these changes and/or apply the given patch first chance I get (probably 24 hours from now)

In the interim I've given [@BillyONeal](https://github.com/BillyONeal) write access to my vcpkg port

**Comment by SamuelMarks** [comment]
I'll manually make these changes and/or apply the given patch first chance I get (probably 24 hours from now)

In the interim I've given [@BillyONeal](https://github.com/BillyONeal) write access to my vcpkg fork

**Review by BillyONeal — DISMISSED** [review]

**Comment by BillyONeal** [comment]
@SamuelMarks I had to make significant changes in order to get this to pass in CI, because this stealth-merge-conflicted with https://github.com/microsoft/vcpkg/pull/26145 which depends on headers this was no longer installing. Please double check that you are happy with these changes.

**Review by BillyONeal — DISMISSED** [review]

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
@SamuelMarks, Please double check that you are happy with these changes.

**Comment by SamuelMarks** [comment]
Yes I'm sure this is fine

**Comment by BillyONeal** [comment]
Thanks for your contribution and thanks for confirming. I resolved the merge conflict and will merge as soon as PR build is green. Thanks!

**Pull request merged**